### DISCUSSION TITLE
expire_build_submitted_for_review should also filter by build platform

### DISCUSSION THREAD

**Discussion by VIadislavJevremovic** [codemagic-ci-cd/codemagic-docs#2495] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

I used the following snippet for TestFlight distribution in codemagic.yaml:

```
app_store_connect:
  auth: integration
  submit_to_testflight: true
  expire_build_submitted_for_review: true
```

There are 2 simultaneous workflows for each released build version, one for iOS target, one for macOS.

### Expected results

Builds are deployed to TestFlight without expiring the same version on a **different** platform.

### Actual results

Builds are deployed to TestFlight but the workflow (either iOS or macOS) which takes longer finds the counterpart build (on the other platform) with the SAME build number and expires it.

```
Expire previous build in review or waiting for review
Found 1 Build matching specified filters: app=APP_ID, expired=False, betaAppReviewSubmission.betaReviewState=WAITING_FOR_REVIEW,IN_REVIEW.
Modify Build BUILD_UUID_TO_EXPIRE
Successfully modified Build BUILD_UUID_TO_EXPIRE
-- Build --
Id: BUILD_UUID_TO_EXPIRE
Type: builds
Expired: True
```

I think the filter needs to include version number + platform as well (to prevent expiring the same version on a **different** platform).

**Comment by himesh-cm** [comment]
Hey @VladyslavBondarenko, Apologies for the delayed reply; the thread got missed.

I will share the feedback with the team, meanwhile Please consider Separate the workflows for iOS and macOS builds to ensure they don't interfere with each other during deployment.

Use distinct build numbers for iOS and macOS builds to avoid conflicts when submitting to TestFlight.

**Comment by VIadislavJevremovic** [comment]
Hello!

Thanks for the reply, no worries.

Workflows for iOS and macOS are indeed separate but the builds NEED to be the same version as some end-users install them on iOS devices and others use macOS Catalyst variant of the same codebase.

**Reply by himesh-cm** [reply]
Hi @VIadislavJevremovic,

If you want to keep the versions the same, managing it via `--build-name=1.0.$BUILD_NUMBER` may not always work. One option is to have the same workflow and fail the build if one of them does not build. Alternatively, you can manage the version as per your convenience via the `pubspec.yaml` file.

Let me know if you still have any questions.

**Reply by VIadislavJevremovic** [reply]
Hello @himesh-cm, I'm not sure you've understood my question altogether.

`expire_build_submitted_for_review` rule should expire the same version number but for THE SAME platform only.
There can be an iOS build and a macOS build on TestFlight with same version numbers.

Why should the workflow making an iOS build expire a macOS build with the same number?

**Comment by mikhail-tokarev** [comment]
Hey @VIadislavJevremovic 

Thanks for reporting this, seems indeed a bug on our side. I don't have any ETA at this moment, but will let you know once I have more information. Sorry for the inconvenience!

**Reply by VIadislavJevremovic** [reply]
Hello @mikhail-tokarev, thank you for taking this into consideration, much appreciated.

**Reply by mikhail-tokarev** [reply]
Hey again! The fix has beed deployed. Can you give it another try?

**Reply by VIadislavJevremovic** [reply]
Hello! Thank you for a quick reply and fix, I will try it out soon and report back.

**Reply by VIadislavJevremovic** [reply]
Hello @mikhail-tokarev @himesh-cm just confirmed that fix works, thanks again! 👍
